File: src/main/java/com/es2/vadebicicleta/externo/email/dto/RequisicaoEmailConverter.java
package com.es2.vadebicicleta.externo.email.dto;

import com.es2.vadebicicleta.externo.email.model.RequisicaoEmail;
import org.springframework.stereotype.Component;

@Component
public class RequisicaoEmailConverter {
    public RequisicaoEmail inDtoToOject(RequisicaoEmailInDto dto){
        return RequisicaoEmail.builder()
                .email(dto.getEmail())
                .assunto(dto.getAssunto())
                .mensagem(dto.getMensagem())
                .build();
    }

    public RequisicaoEmailInDto objectToInDto(RequisicaoEmail object){
        return RequisicaoEmailInDto.builder()
                .email(object.getEmail())
                .assunto(object.getAssunto())
                .mensagem(object.getMensagem())
                .build();
    }
}

File: src/main/kotlin/com/es2/vadebicicleta/externo/email/dto/RequisicaoEmailConverter.kt
package com.es2.vadebicicleta.externo.email.dto

import com.es2.vadebicicleta.externo.commons.UnprocessableEntityException
import com.es2.vadebicicleta.externo.email.model.RequisicaoEmail
import org.springframework.stereotype.Component

@Component
class RequisicaoEmailConverter {
    fun inDtoToOject(dto: RequisicaoEmailInDto): RequisicaoEmail {
        if(dto.email == null || dto.assunto == null || dto.mensagem == null){
            throw UnprocessableEntityException()
        }
        return RequisicaoEmail(dto.email, dto.assunto, dto.mensagem)
    }

    fun objectToInDto(o: RequisicaoEmail): RequisicaoEmailInDto {
        return RequisicaoEmailInDto(o.email, o.assunto, o.mensagem)
    }
}
